src/components/flashMessage.js

```diff
@@ -0,0 +1,41 @@
+import React, { useState, useEffect } from 'react';
+import Bus from '../lib/bus';
+
+const FlashMessage = () => {
+  const intialMessage = { content: '', type: '' };
+  const [message, setMessage] = useState(intialMessage);
+
+  useEffect(() => {
+    Bus.addListener('flash', ({ content, type }) => {
+      displayMessage(content, type);
+    });
+
+    // remove all the listeners from EventEmitter(Bus) so we don't have a memory leak!!
+    return () => {
+      Bus.removeAllListeners('flash');
+    };
+  }, []);
+
+  useEffect(() => {
+    const timeoutId = setTimeout(() => {
+      if (message.content) setMessage(intialMessage);
+    }, 2000);
```

Comment:
What do you think about giving more semantics to the timeout time [2000]? We usually call this **Magic numbers**,  https://en.wikipedia.org/wiki/Magic_number_(programming)

Reply:
Thanks a lot for the valuable suggestions :))

I wonder why it's called magic number it's just a constant?! This industry is filled with buzzwords 😅